File: src/main/java/com/example/jwttest/dto/AuthorityDto.java
package com.example.jwttest.dto;


public class AuthorityDto {
    private String authorityName;

    public AuthorityDto(String authorityName) {
        this.authorityName = authorityName;
    }

    public AuthorityDto() {
    }

    public static AuthorityDtoBuilder builder() {
        return new AuthorityDtoBuilder();
    }

    public String getAuthorityName() {
        return this.authorityName;
    }

    public void setAuthorityName(String authorityName) {
        this.authorityName = authorityName;
    }

    public static class AuthorityDtoBuilder {
        private String authorityName;

        AuthorityDtoBuilder() {
        }

        public AuthorityDtoBuilder authorityName(String authorityName) {
            this.authorityName = authorityName;
            return this;
        }

        public AuthorityDto build() {
            return new AuthorityDto(authorityName);
        }

        public String toString() {
            return "AuthorityDto.AuthorityDtoBuilder(authorityName=" + this.authorityName + ")";
        }
    }
}

File: src/main/kotlin/com/example/jwttest/dto/AuthorityDto.kt
package com.example.jwttest.dto

import com.example.jwttest.entity.Authority

data class AuthorityDto(
    var authorityName: String?
){

    /**
     * AuthorityDto 를 Authority 엔티티로 변환하는 메서드
     * 현재는 안쓰인다.
     */
    companion object {
        fun from(authority: Authority): AuthorityDto {
            return authority.run {
                AuthorityDto(
                    authorityName = authorityName
                )
            }
        }
    }
}